### PR TITLE
Added OCaml version

### PR DESCRIPTION
**Pull request by 314eter**

### CODE REVIEW

File: ocaml/idiomatic.ml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+let empty = None
+
+let singleton value = {
+  x = value;
+  y = Random.int 0x3fffffff;
+  left = None;
+  right = None
+}
+
+let rec has_value tree value =
```
Comment (frol) [inline ocaml/idiomatic.ml]: This implementation does not match all the other implementation. Such benchmark is unfair.

Comment (314eter) [reply ocaml/idiomatic.ml]: That's why I didn't make it the main version, but I didn't realize the benchmark algorithm is that much slower.

**Comment by 314eter** [comment]
The immutable version is comparable to the Haskell implementation. It's slower, but safer and more idiomatic.

**Comment by frol** [comment]
Thank you!

**Comment by frol** [comment]
I have just updated the scoreboards!

**Pull request merged**